cs/src/core/Allocator/PendingFlushList.cs

```diff
@@ -1,54 +1,43 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT license.
 
-using System.Threading;
+using System.Collections.Generic;
 
 namespace FASTER.core
 {
-    class PendingFlushList
+    sealed class PendingFlushList
     {
-        const int maxSize = 8;
-        const int maxRetries = 10;
-        public PageAsyncFlushResult<Empty>[] list;
+        public readonly LinkedList<PageAsyncFlushResult<Empty>> list;
 
         public PendingFlushList()
         {
-            list = new PageAsyncFlushResult<Empty>[maxSize];
+            list = new();
         }
 
-        public bool Add(PageAsyncFlushResult<Empty> t)
+        public void Add(PageAsyncFlushResult<Empty> t)
         {
-            int retries = 0;
-            do
+            lock (list)
```

Comment:
Perf idea: let's reuse the original performant code path with a fixed size array --- if array full, grow the array under critical section. For concurrency, can use something similar to SimpleVersionScheme in libDPR to ensure minimal overhead on the normal code path.

Reply:
Growing is fine, but then we would have to shrink at some point. Also, the growability implies we still have to take the lock every time.

Reply:
I was also thinking of a LinkedList, where each element is the performant array from earlier. Kind of like the ElasticCircularBuffer ...

Reply:
We don't really need to shrink here? Even a thousand pending flush entry is nothing in terms of mem size. 

For the lock --- we can use epoch + one atomic variable to avoid a shared lock on the common path. SimpleVersionScheme does this.